### DISCUSSION TITLE
I wonder the difference between the two Settings

### DISCUSSION THREAD

**Discussion by DYLDYLDYL** [idaholab/moose#23155] [Q&A Modules: Solid mechanics] [answered]
Hi！

I wonder the difference between the two Settings.  The former I learn from Examples and Tutorials, the latter I learn from Combined Examples.
And if  I use the latter, how to output vonmises_stress?

```
[Modules/TensorMechanics/Master]
  [all]
    block = 1
    add_variables = true
    strain = SMALL
    automatic_eigenstrain_names = true
    generate_output = 'vonmises_stress'
  []
    [all_2]
    block = 2
    add_variables = true
    strain = SMALL
    automatic_eigenstrain_names = true
    generate_output = 'vonmises_stress'
  []
```

```
  [TensorMechanics]
    displacements = 'disp_x disp_y disp_z'
  []

  [strain_inner] #We use small deformation mechanics
    type = ComputeSmallStrain
    displacements = 'disp_x disp_y disp_z'
    eigenstrain_names = 'ther_expansion solid_expansion'
    block = 1
  []
  [strain_outer] #We use small deformation mechanics
    type = ComputeSmallStrain
    displacements = 'disp_x disp_y disp_z'
    eigenstrain_names = 'ther_expansion_2 zr'
    block = 2
  []
```

Thanks in advance for your help!

**Comment by GiudGiud** [comment, marked as answer]
Hello

the first one is called an Action, the TensorMechanics master action, and it’s a shorter syntax that behind the hood creates much of the second syntax. The second syntax is specifying objects more manually. 

Using actions is less prone to user errors. 

You can output material properties by specifying in the Material block that defines them which output you would like to place them in. 

Guillaume

**Reply by hugary1995** [reply]
Guillaume is correct. If you want to see which objects get created by the action, add these options to the Problem block:
```python
[Problem]
  type = DumpObjectsProblem
  dump_path = Modules/TensorMechanics/Master/all
[]
```

Once you do that, you will see that [RankTwoScalarAux](https://mooseframework.inl.gov/source/auxkernels/RankTwoScalarAux.html) is responsible for computing the vonmises stress.

**Reply by DYLDYLDYL** [reply]
Thanks!
